### PR TITLE
Document the Rust API.

### LINKED ISSUE #13: Documentation
Documentation can be written and automatically generated using `cargo doc`. It is handled in-source using triple-slash comments, for example: <br/>
`/// this trait does trait stuff`
Would be helpful to document this for contract devs.

### PR DESCRIPTION
**Pull request by jakelang**
A documentation site can now be automatically generated from the source-level docs using `cargo doc`. This can then be hosted using Github Pages.

Closes #13

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -86,32 +88,41 @@ fn unsafe_alloc_buffer(len: usize) -> Vec<u8> {
     ret
 }
 
+/// Enum representing an error code for `codeCopy`, `callDataCopy`, `externalCodeCopy`, and
```
Comment (axic) [inline src/lib.rs]: I would change this to: Enum representing an error code. Currently used by ...

Comment (jakelang) [reply src/lib.rs]: Fixed.

File: src/lib.rs [hunk 2]
```diff
@@ -133,6 +146,7 @@ pub fn external_balance(address: &[u8; 20]) -> [u8; 16] {
     ret
 }
 
+/// Returns the beneficiary address for the most recent block mined.
```
Comment (axic) [inline src/lib.rs]: I would rephrase these to read "the block this transaction is in (current block)"

File: src/lib.rs [hunk 3]
```diff
@@ -190,6 +211,7 @@ pub fn tx_origin() -> [u8; 20] {
     ret
 }
 
+/// Appends log data to the transaction receipt, with a variable number of topics.
```
Comment (axic) [inline src/lib.rs]: Does this needs a documentation if it is not public? Will be included in the external documentation?

File: src/lib.rs [hunk 4]
```diff
@@ -211,6 +233,7 @@ fn log(
     }
 }
 
+/// Appends log data with zero topics.
```
Comment (axic) [inline src/lib.rs]: `without a topic`?

**Comment by axic** [comment]
I am a bit torn, should this describe methods in terms of EEI methods (and refer to their names) or duplicate the high level descriptions from the EEI spec?
